lib/mutant/integration/minitest.rb

```diff
@@ -0,0 +1,114 @@
+require 'minitest'
+
+# monkey patch for minitest
+module Minitest
+  # prevent autorun from running tests when the vm closes
+  def self.autorun
+  end
+end
+
+module Mutant
+  class Integration
+    # Minitest integration
+    class Minitest < self
+      # represents the data associated with a test
+      class TestCase
+        attr_accessor :class_name
+        attr_accessor :test_method
+
+        def initialize(class_name, test_method)
+          @class_name = class_name
+          @test_method = test_method
+        end
+
+        def clazz
+          Object.const_get(class_name)
+        end
+
+        def method_name
+          test_method[5..-1]
+        end
+
+        def runs_successfully?(reporter)
+          ::Minitest::Runnable.run_one_method(clazz, test_method, reporter)
+          reporter.passes?
+        end
+      end
+
+      ALL = Mutant::Expression.parse('*')
+      EXPRESSION_DELIMITER = ' '.freeze
+
+      register 'minitest'
+
+      def initialize
+        @output = StringIO.new
+        @reporter = make_reporter(@output)
+      end
+
+      def setup
+        $LOAD_PATH << 'test' unless $LOAD_PATH.include?('test')
+        $LOAD_PATH << 'lib' unless $LOAD_PATH.include?('lib')
+        Dir.glob('./test/**/*_test.rb').each { |f| require f }
+        Dir.glob('./test/**/test_*.rb').each { |f| require f }
+      end
+
+      # summary reporter detects failures
+      # progress reporter prints dots
+      # could include option verbose if desired
+      def make_reporter(output)
+        options = { io: output }
+        reporter = ::Minitest::CompositeReporter.new
+        reporter << ::Minitest::SummaryReporter.new(options[:io], options)
```

Comment:
Instead to access the hash key `:io`, lets just pass in the lvar `output` as the first argument.